### PR TITLE
[fork] Use AnyInvocable for fork handlers

### PR DESCRIPTION
**Pull request by eugeneo**

### CODE REVIEW

File: src/core/lib/event_engine/posix_engine/ev_poll_posix.cc [hunk 1]
```diff
@@ -323,6 +323,7 @@ bool InitPollPollerPosix() {
   }
   if (grpc_core::Fork::Enabled()) {
     if (grpc_core::Fork::RegisterResetChildPollingEngineFunc(
+            reinterpret_cast<void*>(ResetEventManagerOnFork),
```
Comment (eugeneo) [inline src/core/lib/event_engine/posix_engine/ev_poll_posix.cc:326]:
```suggestion
            reinterpret_cast<void*>(&ResetEventManagerOnFork),
```

File: src/core/lib/iomgr/ev_poll_posix.cc [hunk 2]
```diff
@@ -1405,6 +1405,7 @@ const grpc_event_engine_vtable grpc_ev_poll_posix = {
       }
       if (grpc_core::Fork::Enabled()) {
         if (grpc_core::Fork::RegisterResetChildPollingEngineFunc(
+                reinterpret_cast<void*>(reset_event_manager_on_fork),
```
Comment (eugeneo) [inline src/core/lib/iomgr/ev_poll_posix.cc:1408]:
```suggestion
                reinterpret_cast<void*>(&reset_event_manager_on_fork),
```

**Review by A1exKH — CHANGES_REQUESTED** [review]
@eugeneo, Could you add description for your PR to make it clearer and easier to review, please?

**Comment by eugeneo** [comment]
This PR went nowhere :) Deleting.

**Pull request closed without merging**